src/ops/ops.go

```diff
@@ -917,3 +929,166 @@ func installerArgs(ignitionPath string, device string, extra []string) []string
 	}
 	return append(allArgs, device)
 }
+
+func (o *ops) getPointedIgnitionSource(ignitionPath string) (string, error) {
+	b, err := os.ReadFile(ignitionPath)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to read ignition file %s", ignitionPath)
+	}
+	conf, _, err := config_latest.ParseCompatibleVersion(b)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to parse ignition file %s", ignitionPath)
+	}
+	source := ""
+	for i := range conf.Ignition.Config.Merge {
+		r := &conf.Ignition.Config.Merge[i]
+		if r.Source != nil {
+			source = swag.StringValue(r.Source)
+			if source != "" {
+				break
+			}
+		}
+	}
+	if source == "" {
+		return "", errors.Errorf("source not found for ignition %s", ignitionPath)
+	}
+	return source, nil
+}
+
+func (o *ops) getEmbeddedIgnition(source string) ([]byte, error) {
+	b, err := base64.StdEncoding.DecodeString(strings.Replace(source, "data:;base64,", "", 1))
```

Comment:
`data:blablabla` is actually a URL, with scheme `data`, you can use a dataurl library to parse it:

https://pkg.go.dev/github.com/vincent-petithory/dataurl

We use it in assisted-service so you can see how it's being used there